### PR TITLE
Improve exports, and make contract and web3 lazily initialized

### PR DESCRIPTION
**Pull request by anxolin**
This PR makes the `web3` creation to be done using a constructor function that it's exposed in the library.

This way, we can expose the constructor and not the instance. 
An example of how a client of the library might use these constructors for the contracts and web3 objects can be found in the sandboxes.

See:
* `getOrderPlacements` sandbox
* `test/helpers/contracts.ts` 
* `test/helpers/web3.ts`

Related PRs in (use this changes):
- Telegram (we get rid of the `@gnosis.pm/dex-js/build/src/bla/bla` routes): https://github.com/gnosis/dex-telegram/pull/77
- Web (not so many changes, but gets rid of casting the ABI and simplifying it's consumption): https://github.com/gnosis/dex-react/pull/360

### CODE REVIEW

File: src/contracts/index.ts [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
-  assert(STABLE_COIN_CONTRACT_ADDRESS, 'STABLE_COIN_CONTRACT_ADDRESS env is required')
-  const stableCoinContractAddress = STABLE_COIN_CONTRACT_ADDRESS as string
+export function getBatchExchangeContract (web3: Web3, address?: string): BatchExchangeContract {
+  const batchExchangeAddress = address || process.env.STABLE_COIN_CONTRACT_ADDRESS
 
+  assert(batchExchangeAddress, 'address param, or STABLE_COIN_CONTRACT_ADDRESS env is required')
   // FIXME: There's an issue with this conversion: https://github.com/gnosis/dex-telegram/issues/14
-  const unknownContract = new web3.eth.Contract(batchExchangeAbi as AbiItem[], stableCoinContractAddress) as unknown
+  const unknownContract = new web3.eth.Contract(
+    batchExchangeAbi as AbiItem[],
```
Comment (Velenir) [inline src/contracts/index.ts]: `as AbiItem[]` shouldn't be necessary

File: src/contracts/index.ts [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+    batchExchangeAddress as string
+  ) as unknown
   return unknownContract as BatchExchangeContract
 }
 
-export function getErc20 (): Erc20Contract {
+export function getErc20Contract (web3: Web3): Erc20Contract {
   // FIXME: There's an issue with this conversion: https://github.com/gnosis/dex-telegram/issues/14
   const unknownContract = new web3.eth.Contract(erc20Abi as AbiItem[]) as any
```
Comment (Velenir) [inline src/contracts/index.ts]: `as AbiItem[]` shouldn't be necessary

File: test/sandbox/web3/getNodeInfo.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { web3 } from 'helpers/web3'
 import Logger from 'helpers/Logger'
+import { web3 } from '../../helpers/web3'
```
Comment (Velenir) [inline test/sandbox/web3/getNodeInfo.ts:2]: web3 is no longer exported from there

Comment (anxolin) [reply test/sandbox/web3/getNodeInfo.ts:2]: That a new web3 helper used only for tests:

image

**Review by Velenir — APPROVED** [review]
Cool!

**Review by alfetopito — APPROVED** [review]

**Pull request merged**